plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsSecurityContext.java

```diff
@@ -0,0 +1,153 @@
+ */
+public class HdfsSecurityContext {
+
+    private static final Logger LOGGER = Loggers.getLogger(HdfsSecurityContext.class);
+
+    private static final Permission[] SIMPLE_AUTH_PERMISSIONS;
+    private static final Permission[] KERBEROS_AUTH_PERMISSIONS;
+    static {
+        // We can do FS ops with only a few elevated permissions:
+        SIMPLE_AUTH_PERMISSIONS = new Permission[]{
+            new SocketPermission("*", "connect"),
+            // 1) hadoop dynamic proxy is messy with access rules
+            new ReflectPermission("suppressAccessChecks"),
+            // 2) allow hadoop to add credentials to our Subject
+            new AuthPermission("modifyPrivateCredentials")
+        };
+
+        // If Security is enabled, we need all the following elevated permissions:
+        KERBEROS_AUTH_PERMISSIONS = new Permission[] {
+            new SocketPermission("*", "connect"),
+            // 1) hadoop dynamic proxy is messy with access rules
+            new ReflectPermission("suppressAccessChecks"),
+            // 2) allow hadoop to add credentials to our Subject
+            new AuthPermission("modifyPrivateCredentials"),
+            // 3) allow hadoop to act as the logged in Subject
+            new AuthPermission("doAs"),
+            // 4) Listen and resolve permissions for kerberos server principals
+            new SocketPermission("localhost:0", "listen,resolve"),
+            // We add the following since hadoop requires the client to re-login when the kerberos ticket expires:
+            // 5) All the permissions needed for UGI to do it's weird JAAS hack
+            new RuntimePermission("getClassLoader"),
+            new RuntimePermission("setContextClassLoader"),
+            // 6) Additional permissions for the login modules
+            new AuthPermission("modifyPrincipals"),
+            new PrivateCredentialPermission("org.apache.hadoop.security.Credentials * \"*\"", "read"),
+            new PrivateCredentialPermission("javax.security.auth.kerberos.KerberosTicket * \"*\"", "read"),
+            new PrivateCredentialPermission("javax.security.auth.kerberos.KeyTab * \"*\"", "read")
+            // Included later:
+            // 7) allow code to initiate kerberos connections as the logged in user
+            // Still far and away fewer permissions than the original full plugin policy
+        };
+    }
+
+    /**
+     * Locates the keytab file in the environment and verifies that it exists.
+     * Expects keytab file to exist at {@code $CONFIG_DIR$/kerberos/krb5.keytab}
+     */
+    public static Path locateKeytabFile(Environment environment) {
+        Path keytabPath = environment.configFile().resolve("repository-hdfs").resolve("krb5.keytab");
+        try {
+            if (Files.exists(keytabPath) == false) {
+                LOGGER.error("Could not locate keytab file at [{}].", keytabPath);
+                throw new RuntimeException("Could not locate keytab");
```

Comment:
Good point. Originally I was logging the path but since it was going to be user configurable at the start I figured it would be wise to not broadcast a potentially sensitive file location to any requestor who asks. Since we're standardizing on one canonical place for the file now I don't see why we can't add it back in.